src/server/game/Maps/Map.cpp

```diff
@@ -678,6 +678,20 @@ void Map::Update(const uint32 t_diff)
         player->Update(t_diff);
 
         VisitNearbyCellsOf(player, grid_object_update, world_object_update);
+
+        // Handle updates for creatures in combat with player and are more than 60 yards away
+        if (player->IsInCombat())
+        {
+            HostileReference* ref = player->getHostileRefManager().getFirst();
+
+            while (ref)
+            {
+                if (Unit* unit = ref->GetSource()->GetOwner())
+                    if (unit->GetTypeId() == TYPEID_UNIT && !unit->IsWithinDist2d(player->PositionXYStream().Pos, 60.0f))
```

Comment:
Good point. I didn't see IsWithinDistInMap. I also think I need to make a check for being on the same map. Generally you're removed from combat when shifting map.. But still.